hadoop-hdds/client/src/main/java/org/apache/hadoop/hdds/scm/storage/BlockInputStream.java

```diff
@@ -199,17 +199,19 @@ private boolean isConnectivityIssue(IOException ex) {
     return Status.fromThrowable(ex).getCode() == Status.UNAVAILABLE.getCode();
   }
 
-  private void refreshPipeline(IOException cause) throws IOException {
+  private void refreshBlockInfo(IOException cause) throws IOException {
     LOG.info("Unable to read information for block {} from pipeline {}: {}",
         blockID, pipeline.getId(), cause.getMessage());
-    if (refreshPipelineFunction != null) {
-      LOG.debug("Re-fetching pipeline for block {}", blockID);
-      Pipeline newPipeline = refreshPipelineFunction.apply(blockID);
-      if (newPipeline == null) {
-        LOG.debug("No new pipeline for block {}", blockID);
+    if (refreshFunction != null) {
+      LOG.debug("Re-fetching pipeline and block token for block {}", blockID);
+      BlockLocationInfo blockLocationInfo = refreshFunction.apply(blockID);
+      if (blockLocationInfo == null) {
+        LOG.debug("No new block location info for block {}", blockID);
       } else {
-        LOG.debug("New pipeline for block {}: {}", blockID, newPipeline);
-        this.pipeline = newPipeline;
+        LOG.debug("New block location info for block {}: {}",
+            blockID, blockLocationInfo);
```

Comment:
`BlockLocationInfo#toString` includes the token.  I think we should keep logging the pipeline only.

Reply:
omg i missed this one. Will add an addendum to it.